datalad/interface/tests/test_add_archive_content.py

```diff
@@ -192,21 +205,35 @@ def test_add_archive_content(path_orig, url, repo_path):
     with chpwd(repo_path):
         # TODO we need to be able to pass path into add_archive_content
         # We could mock but I mean for the API
-        assert_raises(RuntimeError, add_archive_content, "nonexisting.tar.gz") # no repo yet
 
-        repo = AnnexRepo(repo_path, create=True)
-        assert_raises(ValueError, add_archive_content, "nonexisting.tar.gz")
+        # no repo yet
+        assert_raises(NoDatasetFound, add_archive_content, "nonexisting.tar.gz")
+        ds = Dataset(repo_path).create()
+        res = ds.add_archive_content("nonexisting.tar.gz", on_failure='ignore')
+        assert_in_results(res, action='add-archive-content',
+                          status='impossible')
+
         # we can't add a file from outside the repo ATM
-        assert_raises(FileNotInRepositoryError, add_archive_content, opj(path_orig, '1.tar.gz'))
+        res = ds.add_archive_content(Path(path_orig) / '1.tar.gz',
+                                     on_failure='ignore')
+        assert_in_results(
+            res,
+            action='add-archive-content',
+            status='impossible',
+            type="dataset",
+            message="Can not add archive outside of the dataset"
+              )
```

Comment:
nitpicking (hopefully I pasted enough and not too many of spaces):

```suggestion
        )
```